pkg/controller/controller.go

```diff
@@ -257,3 +260,112 @@ func initResource(clientset kubernetes.Interface) error {
 
 	return nil
 }
+
+//RunGC starts garbage collector
+func (c *Controller) RunGC(d time.Duration) {
```

Comment:
There's no need to run this periodically with the new loop.  Just once at startup should be sufficient.

I think we just need to do the same "look for all svc/deploy/cm objects with our ownerref" query *once at startup*, and just add the implied Function to the DeltaFIFO (`informer.GetStore()`) as a fake delete.  The regular workqueue loop will then pick them up and they'll hit the `!exists` case and be deleted as usual.